compiler/aarch64/codegen/BinaryEvaluator.cpp

```diff
@@ -64,6 +64,42 @@ static TR::Register *addOrSubInteger(TR::Node *node, TR::CodeGenerator *cg)
    return trgReg;
    }
 
+static TR::Register *
+addOrSubLongInteger(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::Node *firstChild = node->getFirstChild();
+   TR::Register *src1Reg = cg->evaluate(firstChild);
+   TR::Node *secondChild = node->getSecondChild();
+   TR::Register *trgReg = cg->allocateRegister();
+   bool isAdd = node->getOpCode().isAdd();
+
```

Comment:
Some work can be done here to tighten the use of virtual registers, thereby improving the resulting register assignment.  In particular, you shouldn't be unconditionally creating a virtual for the target register if you can re-use one of the source registers whose live range ends at this instruction.

You need to check the reference count on each source node and if the reference count goes to zero then you can conclude that the live range ends here, and the register produced from evaluating that source node is eligible for re-use as the target register.  If both source registers die at this instruction then you need to make a heuristic choice on which one to re-use (likely just pick one).

Reply:
Ok. I'll check the reference count on the node to know when I can re-use a register.